src/book.js

```diff
@@ -563,122 +565,95 @@ export class Book {
    */
   draw_spine_marks(curPage, sigDetails, position) {
     const w = 5;
+    let startX, startY, endX, endY;
+
+    if (sigDetails.isSigStart) {
+      startX = position.spineMarkTop[0];
+      startY = position.spineMarkTop[1];
+      endX = position.spineMarkTop[0];
+      endY = position.spineMarkTop[1];
+    } else {
+      startX = position.spineMarkBottom[0];
+      startY = position.spineMarkBottom[1];
+      endX = position.spineMarkBottom[0];
+      endY = position.spineMarkBottom[1];
+    }
+
     if (position.rotation == 0) {
-      const start = {
-        x: sigDetails.isSigStart
-          ? position.spineMarkTop[0] - w / 2
-          : position.spineMarkBottom[0] - w / 2,
-        y: sigDetails.isSigStart ? position.spineMarkTop[1] : position.spineMarkBottom[1],
-      };
-
-      const end = {
-        x: sigDetails.isSigStart
-          ? position.spineMarkTop[0] + w / 2
-          : position.spineMarkBottom[0] + w / 2,
-        y: sigDetails.isSigStart ? position.spineMarkTop[1] : position.spineMarkBottom[1],
-      };
-      const drawOpts = {
-        start,
-        end,
-        thickness: 0.5,
-        color: rgb(0, 0, 0),
-        opacity: 1,
-      };
-      console.log(' --> draw this: ', drawOpts);
-      curPage.drawLine(drawOpts);
+      startX -= w / 2;
+      endX += w / 2;
     } else {
-      curPage.drawLine({
-        start: {
-          x: sigDetails.isSigStart ? position.spineMarkTop[0] : position.spineMarkBottom[0],
-          y:
-            (sigDetails.isSigStart
-              ? position.spineMarkTop[1] - w / 2
-              : position.spineMarkBottom[1]) -
-            w / 2,
-        },
-        end: {
-          x: sigDetails.isSigStart ? position.spineMarkTop[0] : position.spineMarkBottom[0],
-          y:
-            (sigDetails.isSigStart
-              ? position.spineMarkTop[1] + w / 2
-              : position.spineMarkBottom[1]) +
-            w / 2,
-        },
-        thickness: 0.25,
-        color: rgb(0, 0, 0),
-        opacity: 1,
-      });
+      startY -= w / 2;
+      endY += w / 2;
     }
+
+    const drawLineArgs = {
+      start: { x: startX, y: startY },
+      end: { x: endX, y: endY },
+      thickness: position.rotation == 0 ? 0.5 : 0.25,
+      color: rgb(0, 0, 0),
+      opacity: 1,
+    };
+
+    console.log(' --> draw this: ', drawLineArgs);
+    curPage.drawLine(drawLineArgs);
   }
 
   draw_cropmarks(currPage, side2flag) {
+    const lineSettings = {
+      opacity: 0.4,
+      dashArray: [1, 5],
+    };
+    let start, end;
+
     switch (this.per_sheet) {
       case 32:
         if (side2flag) {
+          lineSettings.dashArray = [1, 5];
```

Comment:
lowkey cursed that you can do this, lol

Reply:
Set a key of an object to an array? I figured this functions are already hard to understand so they might as well at least take up less space.

Reply:
oh I meant the fact that you can redefined properties of a `const`, haha

Reply:
oh yeah, it's the object reference which is const, not the properties of the object. I guess I don't think about it because i'm so used to pointers in c.